extensions/amp-ad/0.1/amp-ad.js

```diff
@@ -0,0 +1,372 @@
+/**
+Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {IntersectionObserver} from '../../../src/intersection-observer';
+import {getAdCid} from '../../../src/ad-cid';
+import {getIframe, prefetchBootstrap} from '../../../src/3p-frame';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {listenFor, listenForOnce,
+    postMessage} from '../../../src/iframe-helper';
+import {loadPromise} from '../../../src/event-helper';
+import {parseUrl} from '../../../src/url';
+import {adPrefetch, adPreconnect} from '../../../ads/_config';
+import {timer} from '../../../src/timer';
+import {user} from '../../../src/log';
+import {viewerFor} from '../../../src/viewer';
+
+
+/** @private @const These tags are allowed to have fixed positioning */
+const POSITION_FIXED_TAG_WHITELIST = {
+  'AMP-LIGHTBOX': true,
+};
+
+
+
+/**
+ * @type {boolean} Heuristic boolean as for whether another ad is currently
+ *     loading.
```

Comment:
It's a heuristic boolean which still serves as a boolean I believed. Should I change it to a number? asking because I am not the author.